platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/ProblemSeverity.kt

```diff
@@ -42,4 +42,9 @@ enum class ProblemSeverity {
      * A problem produced by a task marked as [notCompatibleWithConfigurationCache][org.gradle.api.Task.notCompatibleWithConfigurationCache].
      */
     Suppressed,
+
+    /**
+     * A problem produced by a task, requested Configuration Cache[degradation][org.gradle.api.internal.ConfigurationCacheDegradationController.requireConfigurationCacheDegradation].
+     */
+    DegradationRequested
```

Comment:
Not really happy with this, but so we avoiding passing flags to the `ProblemsCause` creation, concerned [here](https://github.com/gradle/gradle/pull/33444#discussion_r2121922389)

We need to augment the problems with this data to have it excluded from the regular problem reporting in the end of the build.

Reply:
I think having a new severity is fine, but I'd call it based on its effects, similar to other serverities. To that end, the problem itself does not request degradation -- the degradation is always requested before we issue problems. So the severity is more about how we treat this problem with respect to build failures, console reporting and the CC report.

For now, the only difference with `Suppressed` is that we filter out the problems-from-degraded-task from the console summary and every other aspect seems to be exactly the same. If we can't converge the two severities into one, then we should at least give them related names.

For instance, we can call the new severity `SuppressedSiliently`.